hs_core/models.py

```diff
@@ -1677,6 +1936,18 @@ def get_hs_term_dict(self):
 
         return hs_term_dict
 
+    def check_folder_creation(self, folder_full_path):
```

Comment:
Same comment as before. This needs to be a generic ResourceFile function. Can be refactored later.